flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/strategy/AdaptiveSkewedJoinOptimizationStrategy.java

```diff
@@ -0,0 +1,317 @@
+import org.apache.flink.streaming.runtime.partitioner.ForwardForConsecutiveHashPartitioner;
+import org.apache.flink.streaming.runtime.partitioner.RescalePartitioner;
+import org.apache.flink.streaming.runtime.partitioner.StreamPartitioner;
+import org.apache.flink.table.api.config.OptimizerConfigOptions;
+import org.apache.flink.table.runtime.operators.join.FlinkJoinType;
+import org.apache.flink.table.runtime.operators.join.adaptive.AdaptiveJoin;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.runtime.scheduler.adaptivebatch.util.VertexParallelismAndInputInfosDeciderUtils.computeSkewThreshold;
+import static org.apache.flink.runtime.scheduler.adaptivebatch.util.VertexParallelismAndInputInfosDeciderUtils.median;
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkState;
+
+public class AdaptiveSkewedJoinOptimizationStrategy
+        extends BaseAdaptiveJoinOperatorOptimizationStrategy {
+    private static final Logger LOG =
+            LoggerFactory.getLogger(AdaptiveSkewedJoinOptimizationStrategy.class);
+
+    private static final int LEFT_INPUT_TYPE_NUMBER = 1;
+    private static final int RIGHT_INPUT_TYPE_NUMBER = 2;
+
+    private boolean initialized = false;
+    private Map<Integer, Map<Integer, long[]>> aggregatedProducedBytesByTypeNumberAndNodeId;
+
+    private OptimizerConfigOptions.AdaptiveSkewedJoinOptimizationStrategy
+            adaptiveSkewedJoinOptimizationStrategy;
+    private long skewedThresholdInBytes;
+    private double skewedFactor;
+
+    @Override
+    public boolean maybeOptimizeStreamGraph(
+            OperatorsFinished operatorsFinished, StreamGraphContext context) throws Exception {
+        initialize(context.getStreamGraph().getConfiguration());
+        visitDownstreamAdaptiveJoinNode(operatorsFinished, context);
+
+        return true;
+    }
+
+    @Override
+    void tryOptimizeAdaptiveJoin(
+            OperatorsFinished operatorsFinished,
+            StreamGraphContext context,
+            ImmutableStreamNode adaptiveJoinNode,
+            List<ImmutableStreamEdge> upstreamStreamEdges,
+            AdaptiveJoin adaptiveJoin) {
+        if (!canPerformOptimization(adaptiveJoinNode)) {
+            return;
+        }
+        for (ImmutableStreamEdge edge : upstreamStreamEdges) {
+            BlockingResultInfo resultInfo = getBlockingResultInfo(operatorsFinished, context, edge);
+            checkState(resultInfo instanceof AllToAllBlockingResultInfo);
+            aggregatedProducedBytesByTypeNumber(
+                    adaptiveJoinNode,
+                    edge.getTypeNumber(),
+                    ((AllToAllBlockingResultInfo) resultInfo).getAggregatedSubpartitionBytes());
```

Comment:
Invoking `getAggregatedSubpartitionBytes()` will lead to the the clearing of fine grained stats. I guess it is not expected?

Reply:
> Invoking `getAggregatedSubpartitionBytes()` will lead to the the clearing of fine grained stats. I guess it is not expected?

In the latest master branch, invoking `getAggregatedSubpartitionBytes()` will not clear the fine grained stats.